### DISCUSSION TITLE
Run in background on raspberry pi

### DISCUSSION THREAD

**Discussion by TheWalkingDadNL** [zwave-js/zwave-js-ui#1576] [Q&A]
I'm running running zwavejs2mqtt on a Raspberry Pi and it runs fine. The installation was as instructed in another discussion:

```
git clone https://github.com/zwave-js/zwavejs2mqtt
cd zwavejs2mqtt
yarn install
yarn run build
yarn start
```

When i SSH to the pi i can start the program and start my config. Values are coming in nicely. But when i close my SSH connection the server is also lost. I tryed searching how to run yarn in the background, but the search ended in nothing. Anyone can point me in the right direction how to keep running this in the background (and even after a restart)? 

[Edit]

Little update i found the following link to make a service: 

https://stackoverflow.com/questions/64353905/running-npm-start-from-execstart-in-systemctl-service-file

Services give the status active, but it still doesnt run. If i run the .sh file on my own it seems te be running just fine. 

[Edit 2]

Fixed the problem and the service is running fine now. Found this in the log:

```
Error: The module '/home/pi/zwavejs2mqtt/node_modules/@serialport/bindings/build/Release/bindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 93. This version of Node.js requires
NODE_MODULE_VERSION 83. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).

```

After a NPM rebuild it worked as it should and its running as a service now.

**Comment by robertsLando** [comment]
I suggest you to use pm2 for this :)
